pkg/processor/runtime/runtime.go

```diff
@@ -234,3 +251,58 @@ func (ar *AbstractRuntime) Stop() error {
 	ar.SetStatus(status.Stopped)
 	return nil
 }
+
+// Control Communication interface implementations:
```

Comment:
move to decdicated interface / implementation and maybe even under a specific pkg